### PR TITLE
GM: Chevy Silverado 2020-21 support

### PR DESCRIPTION
**Pull request by JMPZ11**
**Checklist**
- [x] added entry to CarInfo in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [x] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [x] route with openpilot: 38aa7da107d5d252|2022-08-15--16-01-12
- [ ] route with stock system:

### CODE REVIEW

**Comment by sshane** [comment]
Number of `ACCButtons` messages sent at 10 Hz before cruise canceled from gas press: 4, 3, 5, 2, 5, 7, 3, 3, 7, 4, 4, 2.

**Comment by adeebshihadeh** [comment]
> Number of `ACCButtons` messages sent at 10 Hz before cruise canceled from gas press: `4, 3, 5, 2, 5, 7, 3, 3, 7, 4, 4, 2` from a test route. While not always immediate, we can experiment with making it more consistent after merging. The cruise so far has always canceled without fault.

Let's improve this before merging. Some things to try: sending counter, sending checksum, sending multiple messages.

File: selfdrive/car/gm/gmcan.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
 from selfdrive.car import make_can_msg
 
-def create_buttons(packer, bus, button):
-  values = {"ACCButtons": button}
+def create_buttons(packer, bus, idx, button):
+  values = {
+    "ACCButtons": button,
+    # "RollingCounter": idx,
```
Comment (sshane) [inline selfdrive/car/gm/gmcan.py:6]:
```suggestion
    "RollingCounter": idx,
```

File: selfdrive/car/gm/carcontroller.py [hunk 2]
```diff
@@ -112,7 +112,8 @@ def update(self, CC, CS):
       if (self.frame - self.last_button_frame) * DT_CTRL > 0.1:
```
Comment (sshane) [inline selfdrive/car/gm/carcontroller.py:112]:
```suggestion
      if (self.frame - self.last_button_frame) * DT_CTRL > 0.04:
```

File: selfdrive/car/gm/carcontroller.py [hunk 3]
```diff
@@ -112,7 +112,8 @@ def update(self, CC, CS):
...
         if CC.cruiseControl.cancel:
           self.last_button_frame = self.frame
-          can_sends.append(gmcan.create_buttons(self.packer_pt, CanBus.CAMERA, CruiseButtons.CANCEL))
+          idx = (CS.buttons_counter + 0) % 4
+          can_sends.append(gmcan.create_buttons(self.packer_pt, CanBus.CAMERA, idx, CruiseButtons.CANCEL))
```
Comment (sshane) [inline selfdrive/car/gm/carcontroller.py:116]:
```suggestion
          can_sends.append(gmcan.create_buttons(self.packer_pt, CanBus.CAMERA, CS.buttons_counter, CruiseButtons.CANCEL))
```

File: selfdrive/car/gm/values.py [hunk 4]
```diff
@@ -84,6 +85,9 @@ class GMCarInfo(CarInfo):
   CAR.BUICK_REGAL: GMCarInfo("Buick Regal Essence 2018"),
   CAR.ESCALADE_ESV: GMCarInfo("Cadillac Escalade ESV 2016", "Adaptive Cruise Control (ACC) & LKAS"),
   CAR.BOLT_EUV: GMCarInfo("Chevrolet Bolt EUV 2022-23", "Premier/Premier Redline Trim", video_link="https://youtu.be/xvwzGMUA210", footnotes=[], harness=Harness.gm),
+  # TEMP NOTE: ACC is not standard, only optional on LTZ and High Country Trims if you buy "Safety Package II"
+  # https://cdn.dealereprocess.org/cdn/brochures/chevrolet/2020-silverado1500.pdf
```
Comment (adeebshihadeh) [inline selfdrive/car/gm/values.py:89]: what's the note here for? Safety Package II seems accurate

Comment (adeebshihadeh) [reply selfdrive/car/gm/values.py:89]: hmm, the 2022 Silverado ACC package is different. perhaps we should just use "Adaptive Cruise Control" instead. @JasonJShuler do you expect the 2022 silverado will be the same platform or require a new port?

Comment (sshane) [reply selfdrive/car/gm/values.py:89]: @JasonJShuler mentioned that the Suburban/Tahoe went Global B in 2022, but not the Silverado. So 2023 likely might need to be a new platform

Comment (sshane) [reply selfdrive/car/gm/values.py:89]: Actually I think we support the 2022 as well. From @JasonJShuler: `Yes - If the 2023 is out I think it's still included... the Suburban / Tahoe went Global B in 2022, but no the silverado`

Comment (sshane) [reply selfdrive/car/gm/values.py]: 2022 Silverado ACC packages: https://www.chevrolet.com/content/dam/chevrolet/na/us/english/index/shopping-tools/download-catalog/09-pdf/2022-chevrolet-new-silverado-1500-ebrochure.pdf

**Pull request merged**